### DISCUSSION TITLE
Redirect to New Record after Replicate

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#3656] [Ideas]
It would be helpful if there were a method on the ReplicateAction class that would redirect the user to the newly created record, rather than redirecting them back to the original record.

**Comment by danharrin** [comment]
How about `afterReplicaSaved(fn ($replica) => redirect(ResourceName::getUrl('edit', ['record' => $replica])))`

**Reply by intrepidws** [reply]
Yea, that's what I ended up doing but it ends up doing a double notification of sorts. I think because there is a split second where the action finishes and fires the notification, before the redirect takes place.

**Reply by danharrin** [reply]
Interesting... I saw that with someone else earlier as well. @zepfietje could you take a look maybe?

**Reply by zepfietje** [reply]
Hmm, does the replicate action return a Livewire response before it redirects, @danharrin?

**Reply by danharrin** [reply]
The Livewire response contains the redirect URL

**Reply by zepfietje** [reply]
Why though? Can't it redirect immediately?

**Reply by danharrin** [reply]
It's just an AJAX response, so no

**Reply by zepfietje** [reply]
I'm confused, so the replicate action request also isn't a Livewire request? It seems that's where it goes wrong.

**Reply by danharrin** [reply]
It is a Livewire request. When the form is submitted, it sends a Livewire request via AJAX. The AJAX response contains a redirect URL that is used by Livewire's JavaScript to redirect the user.

**Reply by zepfietje** [reply]
So that suggests our check in the `NotificationsServiceprovider` doesn't work as expected?

**Reply by danharrin** [reply]
Potentially

**Reply by intrepidws** [reply]
I found another instance where I get a double notification like this. I have an action with a `url()` method that sends the user to a route. Within that route action, I'm firing off the notification before redirecting the user back to the same page where the button was initially clicked. The old `Filament::notify()` method used to accept an argument for `isAfterRedirect`, which I believe prevented this double notification - but I don't see anything like that with the new Notifications package?

```php
Tables\Actions\Action::make('activate')
    ->label('Set as Active Customer')
    ->url(fn (Customer $record): string => route('customers.activate', $record)),
```

Also, I can't simply use the `action()` method instead of the `url()` method because this action deals with setting session variables.

**Reply by zepfietje** [reply]
The `isAfterRedirect` parameter has been deprecated for some time already, also in the old API. Sending notifications and showing them after a redirect is handled by default now.

**Reply by zepfietje** [reply]
But I'm not quite sure what's different with redirecting from actions vs. after saving a form?

**Reply by danharrin** [reply]
it's the same thing. both happen on the client side

**Reply by zepfietje** [reply]
I see, the issue also occurs when saving a new record. Really thought this worked fine previously...

**Reply by danharrin** [reply]
I've never seen the issue after saving a record, but yeah it worked before. Someone we're not properly detecting `redirect` on the component, or for whatever reasoning the notification manager is still catching the new notification too early

**Reply by zepfietje** [reply]
Weird, right? Can you try saving a new record? It really occurs every time now.

**Reply by zepfietje** [reply]
Fixed by #3693.
